unstructured_inference/inference/layoutelement.py

```diff
@@ -88,7 +90,7 @@ def concatenate(cls, groups: Iterable[LayoutElements]) -> LayoutElements:
             element_probs=np.concatenate(probs),
             element_class_ids=np.concatenate(class_ids),
             element_class_id_map=class_id_map,
-            source=group.source,
+            source=sources[0],
```

Comment:
It will raise an error if `sources` is an empty list (which means that `group.source` is not set for any `group` in `groups`)